README.md
# freoをあれこれカスタマイズ（ファイル更新差替版）

### 各リリースバージョンの説明
- [main：フルバージョン](https://github.com/cccabinet/freo_customize/tree/main)<br>モバイル用テンプレートを除く全ファイルを収納
- [diff：差替版](https://github.com/cccabinet/freo_customize/tree/diff)<br>変更のあったファイルのみを収納
- [admin：管理画面版](https://github.com/cccabinet/freo_customize/tree/admin)<br>管理用関連ファイルのみを収納
- [responsive：レスポンシブ版](https://github.com/cccabinet/freo_customize/tree/responsive)<br>バグ修正、JSプラグインの更新、レスポンシブに対応したモバイル用テンプレートを除く全ファイルを収納
- **responsive_diff：レスポンシブ差替版**（←現在のバージョン）<br>バグ修正やJSプラグインの更新、レスポンシブに対応をして変更のあったファイルのみを収納

## ファイル構成
<details>
<summary>構成の内容を開く</summary>
<pre><code>freo_customize/
├── freo/
│   ├── css/
│   │   ├── admin.css（管理画面用）
│   │   └── iframe.css（インラインフレーム用）
│   ├── images/
│   │   └── tablesorter/（tablesorter用のファイルを差替）
│   ├── js/
│   │   ├── tinymce/ (TinyMCE用のファイルを追加）
│   │   │   ├── langs/ja.js（日本語表記ファイル）
│   │   │   └── plugins/
│   │   │       ├── freomedia/（freoメディア管理プラグイン）
│   │   │       └── freomediaform/（freoメディア登録プラグイン）
│   │   ├── admin.js（管理画面用）
│   │   ├── default.js（初期画面用）
│   │   └── iframe.js（インラインフレーム用）
│   ├── libs/
│   │   ├── freo/
│   │   │   ├── internals/
│   │   │   │   ├── admin/
│   │   │   │   │   ├── entry_form.php（管理画面用エントリー入力ファイル）
│   │   │   │   │   ├── media.php（管理画面用メディア管理ファイル）
│   │   │   │   │   ├── media_delete.php（管理画面用メディア削除ファイル）
│   │   │   │   │   └── page_form.php（管理画面用ページ入力ファイル）
│   │   │   │   ├── comment/post.php（コメント登録用ファイル）
│   │   │   │   ├── file/default.php（ファイル表示用ファイル）
│   │   │   │   └── setup/default.php（セットアップ画面用ファイル）
│   │   │   ├── plugins/
│   │   │   │   ├── config.filemanager.php（ファイル管理プラグインの設定ファイルを追加）
│   │   │   │   ├── display.entry_calender.php（エントリーカレンダー表示プラグインの表示ファイル）
│   │   │   │   ├── display.entry_gallery.php（エントリーギャラリー表示プラグインの表示ファイル）
│   │   │   │   ├── page.entry_gallery.php（エントリーギャラリー表示プラグインのページファイル）
│   │   │   │   └── page.filemanager.php（ファイル管理プラグインのページファイルを追加）
│   │   │   ├── checker.php（PHP仕様確認ファイル）
│   │   │   ├── common.php（共通関数ファイル）
│   │   │   ├── initialize.php（初期化処理ファイル）
│   │   │   ├── pictogram.php（絵文字関数ファイル）
│   │   │   ├── prepare.php（設定編集ファイル）
│   │   │   ├── transfer.php（セッションID自動付加関数ファイル）
│   │   │   └── version.php（バージョン情報ファイル）
│   │   └── smarty/
│   │       └── plugins/（Smarty用プラグインを追加）
│   └── templates/
│       ├── internals/
│       │   ├── admin/
│       │   │   ├── entry_form.html（エントリー管理画面テンプレート）
│       │   │   ├── entry_preview.html（エントリープレビュー表示用テンプレート差替）
│       │   │   ├── footer.html（管理画面フッタテンプレート）
│       │   │   ├── header.html（管理画面ヘッダテンプレート）
│       │   │   ├── iframe_header.html（インラインフレームヘッダテンプレート）
│       │   │   ├── iframe_madia_form.html（インラインフレームメディア登録/編集テンプレート）
│       │   │   ├── information_form.html（インフォメーション管理画面テンプレート）
│       │   │   ├── information_preview.html（インフォメーションプレビュー表示用テンプレート差替）
│       │   │   ├── madia_form.html（メディア登録/編集テンプレート）
│       │   │   ├── page_form.html（ページ管理画面テンプレート）
│       │   │   └── page_preview.html（ページプレビュー表示用テンプレート差替）
│       │   ├── default/default.html（トップ表示用テンプレート）
│       │   ├── file/password.html（ファイル表示パスワード画面テンプレート）
│       │   ├── page/default.html（ページ表示用テンプレート）
│       │   └── view/default.html（個別エントリー表示用テンプレート）
│       ├── plugins/
│       │   ├── filemanager/（ファイル管理プラグインのテンプレートを追加）
│       │   └── setup.html（セットアップ画面のテンプレート）
│       ├── error.html（エラー画面のテンプレート）
│       ├── footer.html（フッタのテンプレート）
│       └── header.html（ヘッダのテンプレート）
│
├── markItUp!/
│   ├── freo/
│   │   ├── js/
│   │   │   ├── admin.js（管理画面差替用）
│   │   │   └── iframe.js（インラインフレーム差替用）
│   │   ├── markitup/（markItUp!関連ファイルを差替・追加）
│   │   └── templates/internals/admin/
│   │       ├── entry_form.html（エントリー登録/編集の差替テンプレート）
│   │       ├── header.html（管理用ヘッダの差替テンプレート）
│   │       ├── information_form.html（インフォメーション登録/編集の差替テンプレート）
│   │       └── page_form.html（ページ登録/編集の差替テンプレート）
│   └── plugins/
│       └── clap/【拍手送信プラグイン】
│           └── templates/plugins/clap/admin_thank_form.html（お礼登録/編集の差替テンプレート）
│
├── plugins/
│   ├── bookmark/templates/plugins/bookmark/【ブックマーク登録プラグイン】
│   │   └── default.html（ブックマーク一覧表示テンプレート）
│   ├── circle/templates/plugins/circle/【サークル管理プラグイン】
│   │   └── form.html（ユーザー用サークル編集テンプレート）
│   ├── clap/templates/plugins/clap/【拍手送信プラグイン】
│   │   ├── admin_text.html（拍手メッセージ管理テンプレート）
│   │   └── admin_thank_form.html（お礼登録/編集テンプレート）
│   ├── entry_tagmanager/【エントリータグ管理プラグイン】
│   │   └── libs/freo/plugins/
│   │        ├── config.entry_tagmanager.php（設定ファイル）
│   │        └── end.entry_tagmanager.php（処理終了後表示ファイル）
│   ├── form/templates/plugins/form/【フォーム管理プラグイン】
│   │   ├── admin_form.html（フォーム登録/編集テンプレート）
│   │   ├── form.html（送信フォーム表示テンプレート）
│   │   └── iframe_header.html（インラインフレームヘッダテンプレート）
│   ├── media_comic/templates/plugins/media_comic/【漫画表示プラグイン】
│   │   └── default.html（漫画表示テンプレート）
│   ├── menu/templates/plugins/menu/【メニュー登録プラグイン】
│   │   └── admin.html（メニュー管理テンプレート）
│   ├── page_calender/【ページカレンダー表示プラグイン】
│   │   └── libs/freo/plugins/display.page_calender.php（表示ファイル）
│   ├── page_tagmanager/【ページタグ管理プラグイン】
│   │   └── libs/freo/plugins/
│   │        ├── config.page_tagmanager.php（設定ファイル）
│   │        └── end.page_tagmanager.php（処理終了後表示ファイル）
│   ├── pixiv_illust/【pixivイラスト表示プラグイン】
│   ├── pixiv_novel/【pixiv小説表示プラグイン】
│   ├── profile/templates/plugins/profile/【プロフィール拡張プラグイン】
│   │   ├── admin_form.html（プロフィール編集テンプレート）
│   │   └── form.html（ユーザー用プロフィール編集テンプレート）
│   └── task/templates/plugins/task/【タスク登録プラグイン】
│       └── admin_view.html（タスク確認テンプレート）
│
├── plugins_catalog/
│   ├── catalog/templates/plugins/catalog/【ショッピングカートプラグイン】
│   │   ├── admin.html（商品管理テンプレート）
│   │   ├── admin_delivery_prefecture_form.html（地域別送料編集テンプレート）
│   │   ├── admin_form.html（商品登録/編集テンプレート）
│   │   ├── iframe_header.html（インラインフレームヘッダテンプレート）
│   │   ├── order.html（商品注文テンプレート）
│   │   └── order_preview.html（商品注文確認テンプレート）
│   └── catalog_order/templates/plugins/catalog_order/【注文管理プラグイン】
│       ├── admin.html（注文管理テンプレート）
│       ├── admin_order_form.html（注文登録/編集テンプレート）
│       ├── admin_order_print.html（納品書印刷テンプレート）
│       ├── admin_user_form.html（注文者編集テンプレート）
│       └── form.html（注文者情報編集テンプレート）
│
└── template_custom/
    └── freo/
        ├── css/
        │   └── default.css（初期画面差替用）
        └── templates/
            ├── internals/
            │   ├── category/default.html（カテゴリ表示用テンプレート）
            │   ├── default/default.html（トップ表示用差替テンプレート）
            │   ├── entry/default.html（ブログ表示用テンプレート）
            │   └── page/gallery.html（ギャラリーのページ用テンプレート）
            ├── header.html（ヘッダの差替テンプレート）
            └── utility_page.html（ギャラリーのページ用ユーティリティテンプレート）
</code></pre>
</details>
    
## フォルダの種類
- **freo**<br>freo（v1.21.0）本体のうち、更新のあった `css/`、`images/`、`js/`、`libs/`、`templates/` のファイルに[ファイル管理プラグイン](https://freo.jp/plugin/filemanager.html)と[Smarty用プラグイン](https://freo.jp/plugin/smarty/index.html)を追加し、[プレビュー表示を訪問者向けの表示](https://freo.jp/document/customize/preview.html)のテンプレートに差替<br>※`freo/css/colorbox.css`、`freo/js/jquery.js`、`freo/js/jquery.colorbox.js`、`freo/js/jquery.tablesorter.js`、`freo/images/colorbox/`、`freo/tinymce/` は不要なので差し替える場合は同ファイルを削除してもOK<br>:bulb:**freo本体を設置したものに上書きして差替えて下さい**
- **markItUp!**<br>[テキストエディタをTinyMCEからmarkItUp!に変更する](https://freo.jp/document/customize/markitup.html)場合のJSやテンプレートの差替ファイル<br>※同梱しているmarkItUp!本体は最新版にし、さらに[拡張版](https://cccabinet.jpn.org/view/102)（一部を除く）を導入済み<br>:bulb:**一旦freo本体の差替版を設置してから上書きして差替えて下さい**（`freo/js/tinymce/` は不要なので削除してもOK）
- **plugins**<br>[freoの公式サイトで配布されているプラグイン](https://freo.jp/plugin/)（ファイル管理プラグインとショッピングカート系プラグインを除く）のテンプレートファイルなど<br>:bulb:**プラグイン本体を設置したものに上書きして差替えて下さい**
- **plugins_catalog**<br>[ショッピングカートプラグイン](https://freo.jp/plugin/catalog.html)と[注文管理プラグイン](https://freo.jp/plugin/catalog_order.html)のテンプレートファイル<br>:bulb:ショッピングカートプラグインはシェアウェアなので、他のプラグインのテンプレートと分けています<br>:bulb:**プラグイン本体を設置したものに上書きして差替えて下さい**
- **template_custom**<br>[freoでサイト全体を管理する](https://freo.jp/document/customize/custom.html)ためのテンプレートファイル、CSSファイル<br>※ギャラリーのページ用ユーティリティテンプレートも追加<br>:bulb:**一旦freo本体の差替版を設置してから上書きして差替えて下さい**

## カスタマイズの内容
### バグの修正やファイルの追加
- GitHubの[freo本体のPRの修正](https://github.com/refirio/freo/pulls)、[freo公式プラグインのPRの修正](https://github.com/refirio/freo-plugins/pulls)も追加
- 追加のバグ修正
  - [カレンダーの祝日定義の変更・延長](https://cccabinet.jpn.org/view/104)<br>エントリーカレンダー表示プラグイン、[ページーカレンダー表示プラグイン](https://freo.jp/plugin/page_calender.html)の祝日定義は2020年までしかないので2050年までに延長<br>※ [【修正】freoをphp7以上でエラーが起こらないように修正](https://github.com/refirio/freo/pull/11)の修正も両プラグインに追加済み
    - 2016年以降、山の日の追加
    - 2019年の改元による休日の追加や天皇誕生日の変更
    - 2020年、2021年の東京五輪開催・延期に伴う祝日の移動
    - ※データ元：100年分の祝日リスト（2021年7月以降の変更に対応）<br>　https://zangyoukeisan.cocolog-nifty.com/blog/2011/09/post-b23f.html
  - 認証フォームのパスワード入力欄の修正<br>パスワード入力欄を `input type="text"` から `input type="password"` に修正
  - [ショッピングカートプラグイン](https://freo.jp/plugin/catalog.html)の商品配送方法・料金などの変更<br>※すでにプラグインを導入済みの場合は[sqlファイルをダウンロード](https://cccabinet.jpn.org/view/106)してインポートしてください
    - 普通郵便に定形外郵便（規格内・規格外それぞれ250gまで）を追加<br>※規格内とは、長辺34cm以内、短辺25cm以内、厚さ3cm以内および重量1kg以内のもの
    - 配送方法に日本郵便の[ゆうメール](https://www.post.japanpost.jp/service/yu_mail/)、[ゆうパケット](https://www.post.japanpost.jp/service/yu_packet/index.html)、[クリックポスト](https://www.post.japanpost.jp/service/clickpost/index.html)【未使用に設定済】、[スマートレター](https://www.post.japanpost.jp/service/smartletter/)【未使用に設定済】を追加（未使用のものは状態を「使用」に変更してから使用してください）<br>※クリックポストは事前に利用者情報の登録手続が必要
    - 配送方法にヤマト運輸の[ネコポス](https://business.kuronekoyamato.co.jp/service/lineup/nekoposu/)【未使用に設定済】と[宅急便コンパクト](https://business.kuronekoyamato.co.jp/service/lineup/takkyubin_compact/index.html)【未使用に設定済】を追加（料金などの設定を自分に合わせてから状態を「使用」に変更して使用してください）
    - ヤマト運輸のメール便（mail_10, mail_20）が、2015年3月で廃止されたので未使用に変更
    - 料金が2019年10月に値上げされたので変更
    - ゆうパック（60サイズ）とゆうパック（80サイズ）の地域別送料を2019年10月に消費税が10%になったのに伴い値上げされたので変更（それぞれ東京都から差し出した場合の基本料金）
    - 支払い方法に[宅急便コレクト](https://business.kuronekoyamato.co.jp/service/lineup/payment_daibiki/?gclid=CjwKCAiAvriMBhAuEiwA8Cs5lQnzAWhgtTZttg-Vi_6pzSILK34zo91YkfOGC9UjzVdJ9Z2nBuZNjhoCZEMQAvD_BwE)（ヤマト運輸の法人向け代金引換サービス）を追加
  - フッタのfreo.jpのリンクURLを http://freo.jp から https://freo.jp に変更
  - ファイル管理プラグインの管理対象除外ディレクトリの変更<br>TinyMCEの入っているフォルダがTinyMCE4で `tiny_mce` から `tinymce` に変更になったので変更
- [ファイル管理プラグイン](https://freo.jp/plugin/filemanager.html)を標準で導入
- freoの公式サイトで配布されている[Smarty用プラグイン](https://freo.jp/plugin/smarty/index.html)を標準で導入
- エントリー・ページ・インフォメーション登録時の[プレビュー表示を訪問者向けの表示](https://freo.jp/document/customize/preview.html)に差替
### JSライブラリの更新とCDN化
- jQuery、ColorBox、tablesorter（非公式フォーク版）、TinyMCE、markItUp!を最新版に更新<br>さらにjQuery、ColorBox、tablesorter、TinyMCEはCDNで読み込み設定済み（旧版のファイルは削除済み）
  - [ColorBoxのレスポンシブ設定](https://cccabinet.jpn.org/view/65)
  - [メディアの挿入ファイルを更新日時順に](https://cccabinet.jpn.org/view/96)
  - [freoにTinyMCE5を導入する](https://cccabinet.jpn.org/view/103)<br>カラーパレットも調整（基本の16カラー+オレンジ+文字色）
  - **TinyMCE6に対応**（v1.1.0）
  - [メディア管理/メディア登録をTinyMCEのプラグインにする](https://cccabinet.jpn.org/view/105)
  - [TinyMCEのモバイル表示に対応](https://cccabinet.jpn.org/view/101)
  - [markItUp!の設定を調整](https://cccabinet.jpn.org/view/102)<br>カラーパレットも調整（基本の16カラー+オレンジ+文字色）
### レスポンシブに対応
- 表示用の画面だけでなく、管理用画面もレスポンシブ表示に対応<br>ビューポート480px以下の場合は、ユーティリティと管理メニュー画面は簡易的にボタンで開閉可能に


CHANGELOG.md
# 変更ログ

詳しくは、各リリースファイルのREADME.mdを参照してください。

## [1.1.2] - 2023-08-15

### カスタマイズの内容

- JSライブラリのバージョン表記を一部省略したのを再度最新バージョンの表記に変更。
- TinyMCE6の設定を調整。

バージョンアップは、以下のファイルを最新版付属のものに差し替えてください。<br>（★：管理画面用バージョンも該当、■：レスポンシブバージョン（差分版も同様）も該当））

- freo/js/admin.js ★■
- freo/js/tinymce/langs/ja.js ★■
- freo/libs/freo/internals/setup/default.php ■
- freo/libs/freo/prepare.php ■
- freo/libs/freo/version.php ■
- freo/templates/header.html ■
- freo/templates/internals/admin/header.html ★■
- freo/templates/internals/admin/iframe_header.html ★■
- markItUp!/freo/templates/internals/admin/header.html ★■
- plugins/form/templates/plugins/form/iframe_header.html ■
- plugins/media_comic/templates/plugins/media_comic/default.html ■
- plugins_catalog/catalog/templates/plugins/catalog/iframe_header.html ■
- plugins_catalog/catalog_order/templates/plugins/catalog_order/admin_order_print.html ★■
- template_custom/freo/templates/header.html ■
- freo/libs/freo/version.php ★■
- （レスポンシブ版、レスポンシブ差替版のみ）freo/libs/freo/checker.php ←カスタマイズ版の表記が抜けていたので修正

## [1.1.1] - 2023-08-07

### カスタマイズの内容

- PHPファイル内のJSライブラリのバージョン表記を一部省略したものに変更。
- TinyMCE6用翻訳ファイルの更新（「ページ区切り」を「続きを読むを挿入」に変更）

バージョンアップは、以下のファイルを最新版付属のものに差し替えてください。<br>（★：管理画面用バージョンも該当、■：レスポンシブバージョン（差分版も同様）も該当））

- freo/js/tinymce/langs/ja.js ★■
- freo/libs/freo/internals/setup/default.php ■
- freo/libs/freo/prepare.php ■
- freo/libs/freo/version.php ■

## [1.1.0] - 2023-08-06

### カスタマイズの内容

- TinyMCE6に対応。
- 他のライブラリもバージョン表記を一部省略したものに変更。

バージョンアップは、以下のファイルを最新版付属のものに差し替えてください。<br>（★：管理画面用バージョンも該当、■：レスポンシブバージョン（差分版も同様）も該当））

- freo/js/admin.js ★■
- freo/js/tinymce/langs/ja.js ★■
- freo/templates/header.html ■
- freo/templates/internals/admin/header.html ★■
- freo/templates/internals/admin/iframe_header.html ★■
- markItUp!/freo/templates/internals/admin/header.html ★■
- plugins/form/templates/plugins/form/iframe_header.html ■
- plugins/media_comic/templates/plugins/media_comic/default.html ■
- plugins_catalog/catalog/templates/plugins/catalog/iframe_header.html ■
- plugins_catalog/catalog_order/templates/plugins/catalog_order/admin_order_print.html ★■
- template_custom/freo/templates/header.html ■
- freo/libs/freo/version.php ★■

## [1.0.3] - 2022-08-16

### カスタマイズの内容

- 制限パスワードのインプットタイプの修正（input type="text" ⇒ input type="password"）
- 拍手プラグインのお礼管理画面でTinyMCE用のメディア登録・メディア管理プラグインを導入したのに、アイコン画像を削除し忘れたので削除
- タグの位置の修正
- セットアップ設定編集ファイルの修正（jquery.jsをCDNに変更したので、設定の変更）
- cssファイルのカスマイズ版表記の修正（CSS Validation Serviceによると、cssファイルで@charset 規則が指定できるのは、スタイルシートの最初の部分のみとのことで、カスタマイズ表記を後に移動）

バージョンアップは、以下のファイルを最新版付属のものに差し替えてください。<br>（★：管理画面用バージョンも該当、■：レスポンシブバージョン（差分版も同様）も該当））

- freo/css/admin.css ★■
- freo/css/common.css ★
- freo/css/error.css
- freo/css/iframe.css ★■
- freo/css/setup.css
- template_custom/freo/css/default.css
- freo/templates/internals/admin/entry_form.html ★■
- freo/templates/internals/admin/iframe_media_form.html ★■
- freo/templates/internals/admin/media_form.html ★■
- freo/templates/internals/admin/page_form.html ★■
- markItUp!/freo/templates/internals/admin/entry_form.html ★■
- markItUp!/freo/templates/internals/admin/page_form.html ★■
- plugins/clap/templates/plugins/clap/admin_thank_form.html ★■
- plugins/circle/templates/plugins/circle/form.html ■
- freo/libs/freo/prepare.php ■
- freo/libs/freo/version.php ★■



## [1.0.2] - 2022-08-07

### カスタマイズの内容

- エントリー登録/ページ登録の状態の表示部分の「非公開」を「未公開」に修正（markItUp!のファイルも含む）
- TinyMCE関連ファイルの更新と修正
  - admin.jsのTinyMCEの設定部分を修正（文字列の区切りをスペース区切りに統一、textcolorプラグインは標準搭載されたので削除）
  - TinyMCE 日本語翻訳ファイルが更新されたので差替（"ページ区切り"を"「続きを読む」を挿入"に修正済み）

バージョンアップは、以下のファイルを最新版付属のものに差し替えてください。<br>
（★：管理画面用バージョンも該当、■：レスポンシブバージョン（差分版も同様）も該当）
- freo/templates/internals/admin/entry_form.html ★
- freo/templates/internals/admin/page_form.html ★
- markItUp!/freo/templates/internals/admin/entry_form.html ★
- markItUp!/freo/templates/internals/admin/page_form.html ★
- freo/js/admin.js ★■
- freo/js/tinymce/langs/ja.js ★■
- freo/libs/freo/version.php ★■



## [1.0.1] - 2022-02-10

### カスタマイズの内容

- ページ管理のページ並び順の表示を元に戻す。 
- 注文管理プラグインの不具合を修正。

バージョンアップは、以下のファイルを最新版付属のものに差し替えてください。<br>
（★：管理画面用バージョンも該当、■：レスポンシブバージョン（差分版も同様）も該当）
- freo/templates/internals/admin/page.html ★
- plugins_catalog/catalog/libs/freo/plugins/page.catalog.php ★■
- freo/libs/freo/version.php ★■



## [1.0.0] - 2022-01-10

### カスタマイズの内容

★：管理画面用バージョンも該当（テンプレートは管理画面用のみ該当）

■：レスポンシブバージョン（差分版も同様）も該当

#### ファイルの収納・追加など
- freoの最新版とfreoの公式サイトで配布されているプラグインを収納 ★■<br>（ショッピングカートプラグインはシェアウェアなので、関連プラグインは他のプラグインと分けて収納）
- freoでサイト全体を管理するためのテンプレート、CSSファイルを別途収納 ■
- テキストエディタをTinyMCEからmarkItUp!に変更する場合のJSやテンプレートの差替ファイルを別途収納  ★■
- ファイル管理プラグインをfreo本体に追加 ★■
- freoの公式サイトで配布されているSmarty用プラグインをfreo本体に追加 ★■
- エントリー・ページ・インフォメーション登録時のプレビュー表示を訪問者向けの表示に差替 ★■
#### バグの修正など
- GitHubのfreo本体のPRの修正、freo公式プラグインのPRの修正を追加 ★■
- カレンダーの祝日定義の変更・延長 ★■
- 認証フォームのパスワード入力欄の修正 ★■
- ショッピングカートプラグインの商品配送方法・料金などの変更 ★■
- フッタのfreo.jpのリンクURLを http://freo.jp から https://freo.jp に変更 ★■
- ファイル管理プラグインの管理対象除外ディレクトリの変更 ★■
#### JSプラグインの更新とCDN化
- jQuery、ColorBox、tablesorter（非公式フォーク版）、TinyMCE、markItUp!を最新版に更新<br>さらにjQuery、ColorBox、tablesorter、TinyMCEはCDNで読み込み設定済み（旧版のファイルは削除済み） ★■
#### レスポンシブに対応
- 表示用の画面だけでなく、管理用画面もレスポンシブ表示に対応 ★■
#### 関連プラグインの設定の追加など
- 各画面のタイトルを設定 ★
- 各テンプレートファイルに関連プラグイン（Smarty用プラグインも含む）を導入したときに反映されるよう調整 ★<br>※該当するプラグインが導入されていないときは表示されません
- 管理画面のフッタにはfreoのバージョンが入るように変更 ★
- 一部のテンプレートファイルにどの項目のコードが分かるようにコメントを追加 ★
- エントリーとページのコメントに表示されるメールアドレスを特殊なコードに変換
- NEW!マークの設定
- ユーティリティの新着画像でフィルターやパスワード制限などで閲覧できない記事に専用の画像を表示
- パスワード認証プラグインと直接リンク防止プラグインの各テンプレートのヘッダとフッタ部分を表示用のテンプレートをインクルードするのではなく直接コードを記載に変更
- 関連するページへ戻るリンク、登録/編集後の確認リンクなどを追加 ★
#### 公式サイトに記載されているカスタマイズによる機能追加
- ネタバレボタンの実装 ★
- 日時のセレクトボックス化と現在日時の一括入力 ★
- 検索プラグインで検索条件を入力していないときに検索結果を表示しない
- ページ一括表示プラグインと兄妹ページ表示プラグインの現在表示しているページのタイトルにはリンクしない
#### freoカスタマイズ集に記載されているカスタマイズなど個人的に機能追加
- ページ送りの修正<br>`ページ 1 ... 3 4 5 ... 7` のように表示 ★
- 投稿者ユーザーの権限を制限 ★
- エントリーとページの管理画面情報の変更 ★
- 管理画面に未承認データの表示 ★
- ユーザー管理を権限順に ★
- エントリーとページなどの登録/編集画面にタグの候補をタグクラウド一覧で表示 ★
- コメント管理画面にコメントを表示 ★
- 編集画面で登録画像を表示 ★
- エントリーギャラリー表示のファイルを限定


freo/js/admin.js
/*********************************************************************

 freo | 管理画面 (2023/08/15)

 Copyright(C) 2009-2023 freo.jp
 customized：cccabinet（https://cccabinet.jpn.org/)

*********************************************************************/

$(document).ready(function() {
	//管理メニュー表示ボタン
	$('.toggle').click(function(){
		$('#menu').slideToggle();
		$(this).toggleClass('open');
	});

	//メディア管理（tablesorter非公式フォーク版）
	$('#media').tablesorter({
		emptyTo: 'top',
		headers: {
			2: { sorter: 'digit' },
			3: { sorter: false },
			4: { sorter: false }
		}
	});

	//メディアアップロード欄追加
	$('#media_add').click(function() {
		$('#media_file').append($('#media_template').html());
	});
	$('#media_template').hide();

	//メディア挿入準備
	$('a.colorbox').click(function() {
		if (tinymce.isIE) {
			tinyMCE.get('tiny_mce').focus();
			tinyMCE.activeEditor.windowManager.bookmark = tinyMCE.activeEditor.selection.getBookmark();
		}
	});

	//公開終了
	if ($('#article_close_set').val() == '0') {
		$('#article_close').hide();
	}
	$('#article_close_set').change(function() {
		if ($(this).val() == '1') {
			$('#article_close').show();
		} else {
			$('#article_close').hide();
		}
	});

	//閲覧制限
	if ($('#article_restriction').val() == 'group') {
		$('#article_password').hide();
	} else if ($('#article_restriction').val() == 'password') {
		$('#article_group').hide();
	} else {
		$('#article_group, #article_password').hide();
	}
	$('#article_restriction').change(function() {
		if ($(this).val() == 'group') {
			$('#article_group').show();
			$('#article_password').hide();
		} else if ($(this).val() == 'password') {
			$('#article_group').hide();
			$('#article_password').show();
		} else {
			$('#article_group, #article_password').hide();
		}
	});

	//検証
	if ($('#option_type').val() != 'text') {
		$('#option_validate').hide();
	}
	$('#option_type').change(function() {
		if ($(this).val() == 'text') {
			$('#option_validate').show();
		} else {
			$('#option_validate').hide();
		}
	});

	//承認確認
	$('a.approve').click(function() {
		return confirm('本当に承認してもよろしいですか？');
	});
	$('form.approve').submit(function() {
		return confirm('本当に承認してもよろしいですか？');
	});

	//削除確認
	$('a.delete').click(function() {
		return confirm('本当に削除してもよろしいですか？');
	});
	$('form.delete').submit(function() {
		return confirm('本当に削除してもよろしいですか？');
	});

	//設定確認
	$('a.config').click(function() {
		return confirm('本当に設定してもよろしいですか？');
	});
	$('form.config').submit(function() {
		return confirm('本当に設定してもよろしいですか？');
	});

	//ColorBox
	var extensions = ['gif', 'GIF', 'jpeg', 'JPEG', 'jpg', 'JPG', 'jpe', 'JPE', 'png', 'PNG'];

	var target = '';
	$.each(extensions, function() {
		if (target) {
			target += ',';
		}
		target += 'a[href$=\'.' + this + '\']';
	});
	$(target).colorbox({ maxWidth:'95%', maxHeight:'95%' });

	$('a.colorbox').colorbox({ width:'80%', height:'80%', iframe:true });
});

//TinyMCE
tinymce.init({
	language: 'ja',
	language_url : freo_path + 'js/tinymce/langs/ja.js',
	selector:'#tiny_mce',
	entity_encoding : 'raw',
	extended_valid_elements: 'iframe[*]',
	height: 500,
	element_format : 'xhtml',
	plugins: 'advlist lists link image charmap preview anchor nonbreaking searchreplace visualblocks code fullscreen insertdatetime media table pagebreak quickbars help',
	external_plugins:{
		'freomedia': freo_path + 'js/tinymce/plugins/freomedia/plugin.min.js',
		'freomediaform': freo_path + 'js/tinymce/plugins/freomediaform/plugin.min.js',
	},
	toolbar: 'bold underline strikethrough | styles | bullist numlist outdent indent | code fullscreen | forecolor backcolor removeformat | link unlink image media hr charmap pagebreak | freomedia freomediaform | searchreplace | undo redo help',
	toolbar_mode: 'sliding',
	menu: {
		tools: { title: 'Tools', items: 'code | freomedia freomediaform' }
	},
	mobile: {
		menubar: true,
		plugins: 'image link unlink media table charmap hr pagebreak nonbreaking anchor insertdatetime advlist lists checklist help quickbars advtable',
		toolbar: 'bullist numlist styleselect'
	},
	quickbars_insert_toolbar: 'freomedia image media quicktable | hr charmap nonbreaking pagebreak',
	quickbars_selection_toolbar: 'bold underline strikethrough styles | quicklink removeformat',
	color_map: [
		'#FFFFFF', 'ホワイト',
		'#C0C0C0', 'シルバー',
		'#808080', 'グレー',
		'#000000', 'ブラック',
		'#FF0000', 'レッド',

		'#800000', 'マルーン',
		'#FFFF00', 'イエロー',
		'#808000', 'オリーブ',
		'#00FF00', 'ライム',
		'#008000', 'グリーン',

		'#00FFFF', 'アクア',
		'#008080', 'ティール',
		'#0000FF', 'ブルー',
		'#000080', 'ネイビー',
		'#FF00FF', 'フクシャ',

		'#800080', 'パープル',
		'#FFA500', 'オレンジ',
		'#222222', '文字色',
	],
	style_formats: [
		{title: '見出し', items: [
			{title: '見出し4', format: 'h4'},
			{title: '見出し5', format: 'h5'},
			{title: '見出し6', format: 'h6'}
		]},
		{title: 'ブロック', items: [
			{title: '段落', format: 'p'},
			{title: '引用', format: 'blockquote'},
			{title: '整形済み', format: 'pre'},
			{title: 'コード', format: 'code'},
		]},
		{title: '配置', items: [
			{title: '左揃え', icon: 'align-left', format: 'alignleft'},
			{title: '中央揃え', icon: 'align-center', format: 'aligncenter'},
			{title: '右揃え', icon: 'align-right', format: 'alignright'},
			{title: '均等揃え', icon: 'align-justify', format: 'alignjustify'}
		]},
	],
	insertdatetime_formats: ['%Y/%m/%d', '%H:%M:%S'],
	content_css: [
		freo_path + 'css/common.css',
	],
	convert_urls: false
});
